src/platform_impl/android/mod.rs

```diff
@@ -202,8 +205,11 @@ impl EventLoop {
                     app.window_event(&self.window_target, GLOBAL_WINDOW, event);
                 },
                 MainEvent::ConfigChanged { .. } => {
-                    let old_scale_factor = scale_factor(&self.android_app);
-                    let scale_factor = scale_factor(&self.android_app);
+                    let old_config = self.config.clone();
+                    self.config = self.android_app.config();
```

Comment:
> except that at least with `NativeActivity` we don't get a new `Configuration` at runtime even if the theme changes (yes, with `uiMode` in `android:configChanges`).

Are you saying that at least `MainEvent::ConfigChanged` is being raised? It is expected that the config doesn't change, because `android-activity`'s `ConfigurationRef` that you're also storing in `winit` here is actually an `Arc<RwLock>`:

https://github.com/rust-mobile/android-activity/blob/0d299300f4120821ae1fcaaf0276129c512c2c96/android-activity/src/config.rs#L9-L19

And when that `MainEvent::ConfigChanged` event is raised, its internal `ndk::configuration::Configuration` is replaced, but the surrounding `ConfigurationRef` remains the same so all the clones you're making here **exactly to check for specific changes** are thrown out of the window:

https://github.com/rust-mobile/android-activity/blob/0d299300f4120821ae1fcaaf0276129c512c2c96/android-activity/src/native_activity/glue.rs#L599-L606

I believe this footgun is at the very least very poorly documented in intent as highlighted in https://github.com/rust-mobile/android-activity/pull/164#discussion_r1713062377. That PR changes some of it but I believe we can do even better to highlight this mistake.

Reply:
Fantastic lol. I threw this up before even running it, and tried it once. I'll see if I can figure that bit out.

Reply:
Not sure this is the only issue though, because I made sure to write my example to run `Window::theme` every time it draws, which means that the configuration is _also_ stale.

Reply:
Sounds familiar. I experimented with `screenLayout` to turn a full activity recreate (which you know isn't functional in Rust until I submit my branches) and despite raising a `ConfigChanged` event, none of the properties actually changed :sweat_smile:

Reply:
Yeah... so I think I'll just push the Window::theme implementation into its own branch for now, and add a documentation caveat until we can figure out why we're not getting a fresh config.

Reply:
In general I'd have loved for the events to be more stateful, both in the public API as well as internal to `android-activity`.